services/submission/src/main/resources/application.yaml

```diff
@@ -36,6 +36,16 @@ services:
       supported-countries: ${SUPPORTED_COUNTRIES}
       # The default 'origin country' when not provided in submission payload
       default-origin-country: ${ORIGIN_COUNTRY:DE}
+      tek-field-derivations:
+        # Map containing the derived values of DSOS (days since onset of symptoms) from TRL (transmission risk level)
+        # The values below are subject to change
+        dsos-from-trl:
+          1: -4
+          3: -3
+          5: -2
+          6: -1
```

Comment:
Isn't 6 the youngest key? So in this case it should be `6: 0` and `8: -1`?

Reply:
These are just dummy values for now, just for testing purpose. The actual mapping values will be provided later, and i think we will need some changes in the derivation logic (currently in SubmissionKeyNormalizer) which is simplistic at the moment and does not take into account the day of the TEK. There are still discussions ongoing about how the mapping should look like, so there will be follow up PRs which refine this.

Reply:
> Codewise looks pretty good to me. Just want to clarify some things that I don't understand on the concept of normalization:
> 
> What I understood of the current flow is:
> 
> 1. Submission service registers a `DiagnosisKeyNormalizer` object to `DiagnosisKeyBuilder`
> 2. Submission receives a payload
> 3. After validations, payload is mapped to our domain class `DiagnosisKey`.
> 4. During this process, the builder calls the `Normalizer` class supplied by Submission
> 5. The `SubmissionKeyNormalizer` will then set the DSOS (if missing) based on the TRL values.
> 
> Is this enough? It seems a bit too simplistic to just set a default if value is missing. Don't we also have to normalize the values set by the Mobile devices also? Or is this going to be done on the Upload? Since the TRL and DSOS calculation is particular to each app (as far as I know).
> 
> I'm also struggling a bit to understand the overlap between normalization and input validation, and why it was done like this for submission. I thought we would need to normalize the keys coming and going to EFGS (upload and download). Where having or not DSOS in our own app is more in our control.
> 
> Maybe I'm missing the bigger picture, but based on your comment that normalization is contextual to each service, I suppose there is more coming on this topic to other services, right?

Your understanding of the flow in this implementation is correct.  Currently this is just a skeleton implementation of the whole normalization topic, in order to have a common ground  for the Download and Submission services to apply their respective (contextual) logic:
* The Download service will register another implementation of the normalizer interface and will apply the logic which translates the field values submitted by other national backends to our own specification.  That's in a follow up PR.

* The Submission normalizer implementation just derives DSOS and TRL from one another, in case they are missing. No other transformations are done. I guess this is where a confusion might arise, because what the current submission normalizer object does is rather input sanitization in some sense. This derivation logic and EFGS transformation rules  have been put under the same "normalization" roof somehow and that is reflected in the implementation. I thought of changing the interface to something along the lines of 'field transformation' to be more generic and include both normalization and sanitization but it didn't stick.

Hope that clarifies some things.